enhancements/machine-api/out-of-tree-provider-support.md

```diff
@@ -41,119 +39,63 @@ superseded-by:
+It's especially important to do this for OpenStack because switching to the external cloud provider fixes many issues and limitations with the in-tree cloud provider, such as it's reliance on [Nova metadata service](https://docs.openstack.org/nova/latest/admin/metadata-service.html). For OpenStack platform, this means the possibility for deploying on provider networks and at the edge.
 
-### User Stories
+### Goals
 
-#### Story 1:
+- Prepare OpenShift to start using CCM instead of deprecated in-tree cloud providers.
+- Provide the means to deploy and manage various CCMs. It is proposed to create a new cluster operator - `cluster-cloud-controller-manager-operator` for this.
+- Define and implement an upgrading path from older OpenShift versions with the in-tree cloud provider to CCM'ed ones.
+- Define and implement a downgrading path from newer OpenShift versions with the CCM to those that use the in-tree cloud provider.
 
-As a cloud developer, I’d like to improve support for cloud features for openshift. I'd like to develop and test new cloud providers with do not ship with default kubernetes distiribution, and require support for external cloud controller manager.
+### Non-Goals
 
-#### Story 2:
+- [CSI driver migration](https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/) is out of scope of this work.
 
-As a developer I'd like to develop, build and release fixes independently from the kubernetes core, and assume they will first land upstream and then will be carried over into openshift destribution with less effort.
+## Proposal
 
-#### Story 3:
+Our main goal is to start using Cloud Controller Manager in OpenShift 4, and make a seamless transition for all currently supported providers that require `cloud provider` interface: OpenStack, AWS, Azure, GCP and vSphere.
 
-We’d like to discuss technical details related to a specific cloud in a SIG meeting with people who are also involved into development in this domain, and that way gain useful insights into the cloud infrastructure, improve the overall quality of our features, stay on top of the new features, and improve the relations with maintainers outside of our company, which nevertheless share with us a common goal.
+To maintain the lifecycle of the CCM we want to implement a cluster operator called `cluster-cloud-controller-manager-operator`, that will handle all administrative tasks: deploy, restore, upgrade, and so on.
 
 ### Implementation Details
 
-The `cloud-controller-manager-operator` implementation will be hosted within openshift repository in https://github.com/openshift/cluster-cloud-controller-manager-operator. That repository will manage `CCM` provisioning for metioned cloud providers.
-
-- `CCM` will be provisioned in the `openshift-cloud-controller-manager` namespace.
-- Operator will be deployed in the `openshift-cloud-controller-manager-operator` namespace.
```

Comment:
I think those parts are essential for the operator structuring. Let's leave it, and discuss how it will manage resources in the post-install time.